core/src/main/java/com/sequenceiq/cloudbreak/core/flow2/chain/UpgradeCcmFlowEventChainFactory.java

```diff
@@ -0,0 +1,44 @@
+package com.sequenceiq.cloudbreak.core.flow2.chain;
+
+import static com.sequenceiq.cloudbreak.core.flow2.cluster.ccm.upgrade.UpgradeCcmEvent.UPGRADE_CCM_EVENT;
+import static com.sequenceiq.cloudbreak.core.flow2.stack.update.userdata.UpdateUserDataEvents.UPDATE_USERDATA_TRIGGER_EVENT;
+
+import java.util.Queue;
+import java.util.concurrent.ConcurrentLinkedQueue;
+
+import javax.inject.Inject;
+
+import org.springframework.stereotype.Component;
+
+import com.sequenceiq.cloudbreak.common.event.Selectable;
+import com.sequenceiq.cloudbreak.domain.stack.Stack;
+import com.sequenceiq.cloudbreak.domain.stack.cluster.Cluster;
+import com.sequenceiq.cloudbreak.reactor.api.event.cluster.upgrade.ccm.UpgradeCcmTriggerRequest;
+import com.sequenceiq.cloudbreak.reactor.api.event.stack.userdata.UpgradeCcmFlowChainTriggerEvent;
+import com.sequenceiq.cloudbreak.reactor.api.event.stack.userdata.UserDataUpdateRequest;
+import com.sequenceiq.cloudbreak.service.stack.StackService;
+import com.sequenceiq.flow.core.chain.FlowEventChainFactory;
+import com.sequenceiq.flow.core.chain.config.FlowTriggerEventQueue;
+
+@Component
+public class UpgradeCcmFlowEventChainFactory implements FlowEventChainFactory<UpgradeCcmFlowChainTriggerEvent> {
+
+    @Inject
+    private StackService stackService;
+
+    @Override
+    public String initEvent() {
+        return FlowChainTriggers.UPGRADE_CCM_CHAIN_TRIGGER_EVENT;
+    }
+
+    @Override
+    public FlowTriggerEventQueue createFlowTriggerEventQueue(UpgradeCcmFlowChainTriggerEvent event) {
+        Stack stack = stackService.getByIdWithTransaction(event.getResourceId());
+        Cluster cluster = stack.getCluster();
+        Queue<Selectable> flowEventChain = new ConcurrentLinkedQueue<>();
+        flowEventChain.add(
+                new UpgradeCcmTriggerRequest(UPGRADE_CCM_EVENT.event(), event.getResourceId(), cluster.getId(), stack.getTunnel(), event.accepted()));
+        flowEventChain.add(new UserDataUpdateRequest(UPDATE_USERDATA_TRIGGER_EVENT.event(), event.getResourceId(), event.getOldTunnel()));
```

Comment:
Sanity check:
This is the same request as in the UPDATE_USERDATA_STATE action in the `UserDataUpdateActions` class below